modules/seqexec/model/src/main/scala/seqexec/model/operations.scala

```diff
@@ -7,123 +7,139 @@ import seqexec.model.enum.Instrument
 import seqexec.model.enum.Instrument._
 
 object operations {
-  // Operations possible at the sequence level
-  sealed trait SequenceOperations
-
-  object SequenceOperations {
-    case object PauseSequence extends SequenceOperations
-    case object StopSequence extends SequenceOperations
-    case object RunSequence extends SequenceOperations
+  sealed trait OperationLevel
+  object OperationLevel {
+    sealed trait Observation extends OperationLevel
+    sealed trait NsCycle extends OperationLevel
+    sealed trait NsNod extends OperationLevel
   }
 
-  // Operations possible at the observation level
-  sealed trait ObservationOperations
-
-  object ObservationOperations {
-    case object PauseObservation extends ObservationOperations
-    case object StopObservation extends ObservationOperations
-    case object AbortObservation extends ObservationOperations
-    case object ResumeObservation extends ObservationOperations
-    // Operations for Hamamatsu
-    case object PauseImmediatelyObservation extends ObservationOperations
-    case object StopImmediatelyObservation extends ObservationOperations
-    case object PauseGracefullyObservation extends ObservationOperations
-    case object StopGracefullyObservation extends ObservationOperations
-  }
+  import OperationLevel._
 
-  sealed trait SupportedOperations {
+  sealed trait Operations[A <: OperationLevel]
+  object Operations {
+    // Operations possible at the observation level
+    case object PauseObservation extends Operations[Observation]
+    case object StopObservation extends Operations[Observation]
+    case object AbortObservation extends Operations[Observation]
+    case object ResumeObservation extends Operations[Observation]
 
-    /**
-      * Sorted list of operations supported at the sequence level
-      */
-    def observationOperations(
-      isObservePaused: Boolean): List[ObservationOperations]
+    // Operations possible for N&S Cycle
+    case object PauseGracefullyObservation extends Operations[NsCycle]
+    case object StopGracefullyObservation extends Operations[NsCycle]
 
-    /**
-      * Sorted list of operations supported at the observation (row) level
-      */
-    def sequenceOperations: List[SequenceOperations]
+    // Operations possible for N&S Nod
+    case object PauseImmediatelyObservation extends Operations[NsNod]
+    case object StopImmediatelyObservation extends Operations[NsNod]
+  }
+
+  sealed trait OperationLevelType[L <: OperationLevel]
+  implicit object ObservationLevel extends OperationLevelType[Observation]
+  implicit object NsCycleLevel extends OperationLevelType[NsCycle]
+  implicit object NsNodLevel extends OperationLevelType[NsNod]
+
+  sealed trait SupportedOperations {
+    def apply[L <: OperationLevel](isObservePaused: Boolean, isMultiLevel: Boolean)
```

Comment:
I this can be simplified a bit with `apply[L <: OperationLevel: OperationLevelType]`
Given that `OperationLevelType` already has the L type, perhaps it can be extracted from the typeclass?